bundles/core/src/main/java/com/adobe/cq/commerce/core/components/internal/servlets/CommerceRedirectServlet.java

```diff
@@ -0,0 +1,105 @@
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+package com.adobe.cq.commerce.core.components.internal.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.Servlet;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.SlingHttpServletResponse;
+import org.apache.sling.api.servlets.HttpConstants;
+import org.apache.sling.api.servlets.ServletResolverConstants;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.adobe.cq.commerce.core.components.services.UrlProvider;
+import com.adobe.cq.commerce.core.components.services.UrlProvider.IdentifierLocation;
+import com.adobe.cq.commerce.core.components.services.UrlProvider.ProductIdentifierType;
+import com.adobe.cq.commerce.core.components.utils.SiteNavigation;
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.PageManager;
+import com.day.cq.wcm.api.PageManagerFactory;
+
+@Component(
+    service = Servlet.class,
+    immediate = true,
+    property = {
+        ServletResolverConstants.SLING_SERVLET_METHODS + "=" + HttpConstants.METHOD_GET,
+        ServletResolverConstants.SLING_SERVLET_RESOURCE_TYPES + "=" + CommerceRedirectServlet.RESOURCE_TYPE,
+        ServletResolverConstants.SLING_SERVLET_SELECTORS + "=" + CommerceRedirectServlet.PRODUCT_SELECTOR + ","
+            + CommerceRedirectServlet.CATEGORY_SELECTOR,
+        ServletResolverConstants.SLING_SERVLET_EXTENSIONS + "=" + CommerceRedirectServlet.EXTENSION
+    })
+public class CommerceRedirectServlet extends SlingSafeMethodsServlet {
+
+    protected static final String PRODUCT_SELECTOR = "cifproductredirect";
+    protected static final String CATEGORY_SELECTOR = "cifcategoryredirect";
+    protected static final String EXTENSION = "html";
+    protected static final String RESOURCE_TYPE = "core/cif/components/structure/page/v1/page";
+
+    private Pair<IdentifierLocation, ProductIdentifierType> productIdentifierConfig;
+    private Page productPage;
+    private Page categoryPage;
+
+    @Reference
+    private UrlProvider urlProvider;
+
+    @Reference
+    PageManagerFactory pageManagerFactory;
+
+    @Override
+    protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response) throws IOException {
+        String suffix = request.getRequestPathInfo().getSuffix();
+        String[] selectors = request.getRequestPathInfo().getSelectors();
+
+        if (suffix == null) {
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Missing redirect suffix.");
+            return;
+        }
+
+        String[] suffixInfo = suffix.substring(1).split("/");
+        if (suffixInfo.length != 1) {
```

Comment:
This can be simplified to `suffix.substring(1).contains("/")` which doesn't require regex internally.